pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -1009,13 +1182,30 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options)
 		return err
 	}
 
+	// index ExternalSecrets based on the target resource (GVK + name)
+	// this lets us quickly find all ExternalSecrets which target a specific non-Secret resource
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &esv1.ExternalSecret{}, indexESTargetResourceField, func(obj client.Object) []string {
+		es := obj.(*esv1.ExternalSecret)
+		if !isNonSecretTarget(es) || !r.AllowNonSecretTargets {
```

Comment:
```suggestion
		if !r.AllowNonSecretTargets || !isNonSecretTarget(es) {
```

I know this is silly but we should only perform the check if the config allows it.